crates/iroha/tests/multisig.rs

```diff
@@ -110,75 +144,111 @@ fn multisig_base(transaction_ttl_ms: Option<u64>) -> Result<()> {
         key.clone(),
-        "value".parse::<Json>().unwrap(),
+        "congratulations".parse::<Json>().unwrap(),
     )
     .into()];
     let instructions_hash = HashOf::new(&instructions);
 
     let proposer = signatories.pop_last().unwrap();
-    let approvers = signatories;
-
-    let args = &MultisigTransactionArgs::Propose(instructions);
-    let propose = ExecuteTrigger::new(multisig_transactions_registry_id.clone()).with_args(args);
+    let mut approvers = signatories.into_iter();
 
+    let propose = MultisigPropose::new(multisig_account_id.clone(), instructions);
     alt_client(proposer, &test_client).submit_blocking(propose)?;
 
-    // Check that the multisig transaction has not yet executed
-    let _err = test_client
-        .query_single(FindAccountMetadata::new(
-            multisig_account_id.clone(),
-            key.clone(),
-        ))
-        .expect_err("key-value shouldn't be set without enough approvals");
-
     // Allow time to elapse to test the expiration
-    if let Some(ms) = transaction_ttl_ms {
+    if let Some(ms) = transaction_ttl_ms_opt {
         std::thread::sleep(Duration::from_millis(ms))
     };
     test_client.submit_blocking(Log::new(Level::DEBUG, "Just ticking time".to_string()))?;
 
-    // All but the first signatory approve the multisig transaction
-    for approver in approvers.into_iter().skip(1) {
-        let args = &MultisigTransactionArgs::Approve(instructions_hash);
-        let approve =
-            ExecuteTrigger::new(multisig_transactions_registry_id.clone()).with_args(args);
+    let approve = MultisigApprove::new(multisig_account_id.clone(), instructions_hash);
+
+    // Approve once to see if the proposal expires
+    let approver = approvers.next().unwrap();
+    alt_client(approver, &test_client).submit_blocking(approve.clone())?;
 
-        alt_client(approver, &test_client).submit_blocking(approve)?;
+    // Subsequent approvals should succeed unless the proposal is expired
+    for _ in 0..(N_SIGNATORIES - 4) {
+        let approver = approvers.next().unwrap();
+        let res = alt_client(approver, &test_client).submit_blocking(approve.clone());
+        match &transaction_ttl_ms_opt {
+            None => assert!(res.is_ok()),
+            _ => assert!(res.is_err()),
+        }
     }
-    // Check that the multisig transaction has executed
+
+    // Check that the multisig transaction has not yet executed
+    let _err = test_client
+        .query_single(FindAccountMetadata::new(
+            transaction_target.clone(),
+            key.clone(),
+        ))
+        .expect_err("instructions shouldn't execute without enough approvals");
+
+    // The last approve to proceed to validate and execute the instructions
+    let approver = approvers.next().unwrap();
+    let res = alt_client(approver, &test_client).submit_blocking(approve.clone());
+    match (&transaction_ttl_ms_opt, &unauthorized_target_opt) {
+        (None, None) => assert!(res.is_ok()),
```

Comment:
nit: (here and below) `res.unwrap()` and `res.unwrap_err()` provide better diagnostics than plain `assert!(res.is_ok())`, since they would print the error message